### DISCUSSION TITLE
Product Serial Number

### DISCUSSION THREAD

**Discussion by fernandolcx** [solidusio/solidus#5761] [New Features or Ideas]
The ideia is based on what we have in **ERPNext** (https://docs.erpnext.com/docs/user/manual/en/serial-no) 
and **Odoo** (https://www.odoo.com/documentation/17.0/applications/inventory_and_mrp/inventory/product_management/product_tracking/serial_numbers.html)

This implements granular control of individual items in stock, distinguished by its Serial Number (SN) of some specific products,
which can be used to:

* Know what supplier that specific item was bought;
* Track warranty;
* Avoid return fraud;
* Identify stolen or missing products in stock;

All transactions regarding to this product (stock transfer, purchase or selling) must have their item SN specified;

**Comment by jarednorman** [comment]
This is a good candidate for an extension! I don't think it needs to be part of core, though.

**Reply by seandawes** [reply]
This could be useful to us in the future for sure. We have some inventory which has a specific serial number we track outside Solidus.

**Comment by fthobe** [comment]
@fernandolcx if you need ERPnext working with solidus and playing nice, we are working on that and I might publish it as open source.

**Comment by fthobe** [comment]
@jarednorman I think meta data would be a solution for this for example.

**Comment by fthobe** [comment]
@seandawes @fernandolcx 
We are all in the same situation with externally tracked S/N. Our approach is to add additional data to line items (among other objects). Maybe you have an opinion or some aspects to add. It's a rather long discussion but might be worth reading. https://github.com/solidusio/solidus/issues/5897

**Comment by seandawes** [comment]
@fthobe Why not just expand the data columns in the inventory units table to just have a serial column. And you populate this when receiving inventory either if it needs to be done manually or via an automated job. Really doesn't require a ton of work. The issue will lie in how you are receiving inventory as it will need to be singular and not +10 or something at a time.

**Reply by fthobe** [reply]
@seandawes 
I do not think this is technical issue, but a process issue. 

- Because if you use it for picking it's properly better to leverage the erp you use that should have precise stock in the first place and should be the single source of truth
- If you want to show the serial number to the customer any field is fine to just shoot the serial number into the line item via meta data

Is there a particular reason why you want to track serial numbers in the store?

**Reply by seandawes** [reply]
For those not using an ERP. Lot of things the ERPs available don't do what you need and are too much $. Especially now when you can just something easily in www.retool.com ERP is just another database with series of tables. And you are going to make the ERP talk back to the store anyway.  In the case of the store you are freezing stock on fulfillment generally anyway so why not freeze and release serialized stock the same way. 

It really just depends on processes but I could easily see the store just doing this with a few minor changes none of which are critical breaking changes.

**Reply by fthobe** [reply]
Technically probably nothing, the issue I see is that the decision where to “freeze” the serial number is one that is very very custom. 

We try to keep as much logic as possible outside of the store (also because there’s a high rate of PRs getting bounced here). 

If you are interested in keeping track of a serial the meta PR is the right place to voice yourself, if the goal is to have very specific stock logic I’d agree with Jared (I don’t know if he’s watching) to keep it outside and extend the stock model.
